### PR TITLE
Packages: Create `output` and `test-utils` pacakges

### LINKED ISSUE #8778: Packages: Move `output` to its own package
## Task Description

The output package is responsible for creating AMP output for a story.

This would help a lot for things like #8692 where story output needs to be  generated outside of the actual editor.

**To-do:**

* [ ] The unit tests operate with certain element types which are not really known here. Move these to new Jest-based integration tests (as they involve multiple packages)

### PR DESCRIPTION
**Pull request by mohdsayed**
## Context

See https://github.com/GoogleForCreators/web-stories-wp/issues/8778

## Summary

As part of the output package organization work, I had to create one more `test-utils` package, so that it can easily be used in `output` package and can be commonly used in `story-editor`, `wp-story-editor` package as well.

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes https://github.com/GoogleForCreators/web-stories-wp/issues/8778

### CODE REVIEW

File: packages/output/package.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{
```
Comment (swissspidy) [inline packages/output/package.json:1]: Nit: we indent these with 2 spaces, not 4.

File: packages/output/package.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        "@googleforcreators/rich-text": "*",
+        "@googleforcreators/units": "*",
+        "@googleforcreators/url": "*",
+        "prop-types": "^15.8.0"
+    },
+    "devDependencies": {
+        "@googleforcreators/element-library": "*",
+	    "@googleforcreators/test-utils": "*",
+        "@testing-library/react": "^12.1.2",
+	    "flagged": "^2.0.1"
```
Comment (swissspidy) [inline packages/output/package.json]: Mixed indentation here

File: packages/test-utils/package.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{
```
Comment (swissspidy) [inline packages/test-utils/package.json:1]: Same here about indentation using 2 spaces

**Comment by swissspidy** [comment]
With this new `output` package we can finally remove the `story-editor` dependency from `wp-dashboard` here (and of course from `package.json`:

https://github.com/GoogleForCreators/web-stories-wp/blob/f976a7154c57a42bc5caf768c2bba5c628deec7f/packages/wp-dashboard/src/api/story.js#L155-L158

That should help reduce dashboard bundle size quite a bit.

Instead of calling `getStoryPropsToSave`, `createStoryFromTemplate` can call `getStoryMarkup` directly.

File: packages/output/src/page.js [hunk 4]
```diff
@@ -21,14 +21,16 @@ import PropTypes from 'prop-types';
 import { generatePatternStyles } from '@googleforcreators/patterns';
 import { PAGE_HEIGHT, PAGE_WIDTH } from '@googleforcreators/units';
 import { StoryAnimation } from '@googleforcreators/animation';
-import { ELEMENT_TYPES } from '@googleforcreators/elements';
+import {
+  ELEMENT_TYPES,
+  StoryPropTypes,
+  isElementBelowLimit,
```
Comment (swissspidy) [inline packages/output/src/page.js:27]: A bit unfortunate that this util has to live here, as it is more related to the editor canvas, but I guess we'll have to live with that for now. Maybe in the future we can get rid of it, as indicated by that TODO further down in this file.

File: packages/output/src/index.js [hunk 5]
```diff
@@ -16,3 +16,5 @@
 
 export { default as OutputStory } from './story';
 export { default as getStoryMarkup } from './utils/getStoryMarkup';
+export { default as isHexColorString } from './utils/isHexColorString';
```
Comment (swissspidy) [inline packages/output/src/index.js]: This would be more fitting for the `patterns` package

**Comment by mohdsayed** [comment]
> With this new `output` package we can finally remove the `story-editor` dependency from `wp-dashboard` here (and of course from `package.json`:
> 
> https://github.com/GoogleForCreators/web-stories-wp/blob/f976a7154c57a42bc5caf768c2bba5c628deec7f/packages/wp-dashboard/src/api/story.js#L155-L158
> 
> That should help reduce dashboard bundle size quite a bit.
> 
> Instead of calling `getStoryPropsToSave`, `createStoryFromTemplate` can call `getStoryMarkup` directly.


We got significant total size change of `-112kb` and `assets/js/chunk-getStoryPropsToSave.js` got removed, but why `assets/js/wp-story-editor.js` size increased by `+179 kB` 🤔

**Comment by swissspidy** [comment]
In webpack we have a shared bundle for common code used by both `wp-dashboard` and `wp-story-editor`. With this dependency on `story-editor` gone from the dashboard, most of that code is now only used by `wp-story-editor`, so that bundle itself got bigger.

**Review by swissspidy — APPROVED** [review]

**Comment by mohdsayed** [comment]
> In webpack we have a shared bundle for common code used by both `wp-dashboard` and `wp-story-editor`. With this dependency on `story-editor` gone from the dashboard, most of that code is now only used by `wp-story-editor`, so that bundle itself got bigger.

@swissspidy  I guessed that but `-98.7 kB` was reduced by removing `assets/js/chunk-getStoryPropsToSave.js`, and `+179 kB` was added to `assets/js/wp-story-editor.js`. Do you know why such a large difference?

**Comment by swissspidy** [comment]
I see. I guess I'd need to utilize webpack bundle analyzer or so to look into it a bit more closely.

**Pull request merged**